### ISSUE TITLE
Can't launch browser through dappeteer.launch(puppeteer) function

### ISSUE THREAD

**Issue by rayGoMoon** [ChainSafe/dappeteer#43]
**Describe the bug**
Hi! I just can't launch the browser through **dappeteer.launch(puppeteer)** function. At the same time, no error messages could be logged in the console. 

But the original **puppeteer.launch({ headless: false })** function is working fine.

**To Reproduce**

1. Installation
`npm install -s @chainsafe/dappeteer`

2.  Creat the test.mjs file

3.  Run
`node test.mjs`

The following code does not work:

```
"use strict";

import puppeteer from "puppeteer";
import dappeteer from "@chainsafe/dappeteer";

const main = async () => {
  const browser = await dappeteer.launch(puppeteer);
  console.log(browser);
}

main();
```

It can not launch the browser and there are no error messages. And the **console.log(browser)** was not executed.

The following code will start the browser properly.

```
"use strict";

import puppeteer from "puppeteer";
import dappeteer from "@chainsafe/dappeteer";

const main = async () => {
  const browser = await puppeteer.launch({ headless: false });
  console.log(browser);
}

main();
```

**System:**
- OS: [window 10 pro x64]
- OS version [21H1]
- NodeJs version [16.10.0, 16.19.1]
- dAppeteer version [2.1.0]

**Comment by BeroBurny** [comment]
@rayGoMoon thanks for reaching out. I not being able to reproduce. It is possible that the node doesn't have write permission? `dappeteer.launch` will download MetaMask inside `node_modules` and lanch puppeteer with it.

**Comment by rayGoMoon** [comment]
@BeroBurny Hi, Bero, thanks for your reply.

In the past two days, I have tried various methods in the direction you pointed out but still could not succeed. It's frustrating for me and I don't know what's going wrong.

First, I set node to always run as system administrator on my Windows 10 Pro x64 system. This approach did not bring any changes to the result.

Then I installed WSL2 on my Windows 10 system and tried to run dappeteer in the Ubuntu 20.04 virtual machine with WSL2. But still, the same result.

So, I thought there must be something wrong with my Windows 10 PC system.

Therefore, I

1. Migrated my work platform from my PC to my laptop.
2. Installed the latest version of VMware Workstation on my laptop. 
3. Installed the latest Ubuntu 21.04 system in the VMware virtual machine.
4. Let the Ubuntu system automatically update to the latest state.
5. `sudo apt-get install chromium-browser` to install the latest chromium browser. Because I read from some posts on the web that installing chromium browser installs the necessary dependencies for puppeteer in advance.
6. Followed the official instructions to install nvm.
7. `nvm install-latest-npm` to install the latest npm v7.24.1.
8. `nvm install node` to install and use the latest nodejs v16.10.0.
9. `npm install -s @chainsafe/dappeteer` to install the dappeteer v2.1.0.
10. Create the same test.mjs file.
11. `node test.mjs` to run.

However, I still got the same failed result.

`await puppeteer.launch({ headless: false })` can successfully launch the browser.

`await dappeteer.launch(puppeteer)` can not launch the browser and there are no error messages. The script just runs silently for a while and then ends automatically.

In addition, during the test, I found a very strange phenomenon. 

Regardless of the operating system, there are times when I run `npm install -s @chainsafe/dappeteer` to install the dappeteer and then the puppeteer doesn't actually get installed successfully. 

In the .local-chromium folder of puppeteer, there is an unzipped chrome-linux.zip archive(or chrome-windows.zip).

![pptr_01](https://user-images.githubusercontent.com/44758368/134873970-a74f9358-a007-4459-b4bd-026af0796f7f.png)

I had to delete all the files and do the installation again. It took several attempts to get the correct result.

![pptr_02](https://user-images.githubusercontent.com/44758368/134874163-7885579c-d526-417a-af43-bd0bccceb3b3.png)

So I think maybe it's because the scripts under the @chainsafe/dappeteer folder are also not fully installed, which is causing all the problems.

So I went into the @chainsafe/dappeteer folder, ran `npm install` and got the following results.

![d_02](https://user-images.githubusercontent.com/44758368/134874334-4a119523-71ed-484d-ba5f-e86faa9da8dd.png)

![d_01](https://user-images.githubusercontent.com/44758368/134874379-d8c11506-7184-4b24-a5de-f093d38cde88.png)

The installation went well, but still the `await dappeteer.launch(puppeteer)` can not launch the browser.

Finally, I used nvm to change the node version to v14.17.6 LTS. But still got the same failed result.

Now I'm really at my wit's end.

**Bero, would you be so kind as to share your dev system setup with me?** Which operating system are you using? Which version of node are you using?

It would be of great help to me, thanks a lot.

**Comment by BeroBurny** [comment]
tested on 2 systems

```
  System:
    OS: Linux 5.8 Ubuntu 20.04.2 LTS (Focal Fossa)
  Binaries:
    Node: 16.10.0 - ~/.nvm/versions/node/v16.10.0/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 7.24.0 - ~/.nvm/versions/node/v16.10.0/bin/npm
```

```
  System:
    OS: Windows 10 10.0.18363
  Binaries:
    Node: 16.10.0 - ~/.nvm/versions/node/v16.10.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 7.24.0 - ~/.nvm/versions/node/v16.10.0/bin/npm
```

did you tried to clone `dappeteer` project and run `npm install` and `npm run test` ?

**Comment by BeroBurny** [comment]
@rayGoMoon 
created this project bases on 1st repo steps https://github.com/BeroBurny/dappeteer-43

cone it and run `npm install`

and after that run
- `node dappeteer_auto.mjs`
- `node dappeteer_manual.mjs`
- `node puppeteer_no_metamask.mjs`
- `node puppeteer_metamask.mjs`

and additional

- `node dappeteer_manual_v2.mjs`
- `node puppeteer_metamask_v2.mjs`

report back situation, probably this will help me to identify if there is some issue inside a code

**Comment by rayGoMoon** [comment]
@BeroBurny Thank you for your reply, I will try it right away and get back to you.

**Comment by rayGoMoon** [comment]
Hi! I tested it on my windows 10 PC, using node v16.10.0, and got the following results.

1. `node dappeteer_auto.mjs`

<img width="449" alt="win_001" src="https://user-images.githubusercontent.com/44758368/134934952-ed1e142d-e266-41ca-91f6-47785b29bbe0.png">

2. `node dappeteer_manual.mjs`

Successfully launched the browser and opened the welcome page of metamask.

<img width="459" alt="win_002" src="https://user-images.githubusercontent.com/44758368/134935107-b074172c-e775-4bbc-8583-5e695a634125.png">

3. `node puppeteer_no_metamask.mjs`

The result is the same as `node dappeteer_manual.mjs`. 

Successfully launched the browser and opened the welcome page of metamask.

<img width="481" alt="win_003" src="https://user-images.githubusercontent.com/44758368/134935290-147a1058-1417-4a8d-ba2a-08814f471b64.png">

4. `node puppeteer_metamask.mjs`

The result is the same as `node dappeteer_manual.mjs` and `node puppeteer_no_metamask.mjs`. 

Successfully launched the browser and opened the welcome page of metamask.

<img width="485" alt="win_004" src="https://user-images.githubusercontent.com/44758368/134935390-84bee6b7-7086-491c-83d6-5ef42fc8ebd3.png">

5. `node dappeteer_manual_v2.mjs`

Unable to start the browser. Throwing an error.

<img width="566" alt="win_005" src="https://user-images.githubusercontent.com/44758368/134935501-3601ea7b-0284-42d2-a54c-88da8236c715.png">

6. `node puppeteer_metamask_v2.mjs`

Unable to start the browser. Throwing an error.

<img width="586" alt="win_006" src="https://user-images.githubusercontent.com/44758368/134935625-024c356e-7ba7-4b44-9a58-c91e94af34b2.png">

**Comment by BeroBurny** [comment]
@rayGoMoon thanks your help is pretty helpful!
I to forget one thing can you delete the content of folder `metamask` and again run `node dappeteer_manual.mjs`

**Comment by BeroBurny** [comment]
@rayGoMoon 
Hey, created new repo focusing on the part where is problem
- `clone` https://github.com/BeroBurny/bb-dappeteer-43
- run `npm install`

after install run commands
- `node dappeteer_auto.mjs`
- `node dappeteer_manual.mjs`

in case you god result output some like this
![image](https://user-images.githubusercontent.com/6376047/135048181-e18a237a-2324-4144-a0a8-164d358b8fe0.png)
please remove `node_module` and `metamask` folders and run `npm install` then try again

**Comment by rayGoMoon** [comment]
Hi, Bero! Thank you so much for your help.

After running and reviewing the test script you provided above, I just realized that the **problem is in my web proxy settings**.

I use a software called [Xray](https://github.com/XTLS/Xray-core) in my OpenWrt router to proxy all internet traffic. Before this, everything was working fine. I could browse the web, watch youtube videos, and use Git and npm perfectly fine.

But this time, for some reason, it causes a network error when metamask is downloaded, which causes the whole software to fail to run.

The same reason seems to have caused the puppeteer to sometimes not install properly.

So, I replaced the web proxy software with [MysteriumVPN](https://mysterium.network/apps/), and this time all the code worked fine.

I also tested it with your newly released version 2.2.0 software and it also works fine.

**Thank you again for all the kind help and time you have given to me.**

Below are the test results.

`git clone https://github.com/BeroBurny/bb-dappeteer-43`

`npm install`

**Using Xray in my OpenWrt router to proxy all internet traffic：**

❌1. `node dappeteer_auto.mjs`

<img width="644" alt="bb_001" src="https://user-images.githubusercontent.com/44758368/135092125-2b9bd667-92f7-4dba-8574-6f195a116f15.png">

❌2. `node dappeteer_manual.mjs`

<img width="646" alt="bb_002" src="https://user-images.githubusercontent.com/44758368/135092218-935fb4aa-0b26-4673-a2dd-75d079349f22.png">

**Using MysteriumVPN to proxy all internet traffic：**

✔️1. `node dappeteer_auto.mjs`

Successfully launched the browser and opened the welcome page of metamask.

<img width="647" alt="bb_005" src="https://user-images.githubusercontent.com/44758368/135092615-75ecdf56-0264-464c-89e7-97ca6e1d17c3.png">

✔️2. `node dappeteer_manual.mjs`

Successfully launched the browser and opened the welcome page of metamask.

<img width="650" alt="bb_006" src="https://user-images.githubusercontent.com/44758368/135092706-e5807523-16cf-4924-9ea6-d1209702775d.png">

**Comment by BeroBurny** [comment]
Keep it open as need to improve logging in case of future issues
